File: gto-support-db/src/main/java/org/ccci/gto/android/common/db/WalSQLiteOpenHelper.java
package org.ccci.gto.android.common.db;

import android.content.Context;
import android.database.DatabaseErrorHandler;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.NonNull;

public abstract class WalSQLiteOpenHelper extends SQLiteOpenHelper {
    public WalSQLiteOpenHelper(@NonNull final Context context, final String name,
                               final SQLiteDatabase.CursorFactory factory, final int version) {
        super(context, name, factory, version);
        initWal();
    }

    public WalSQLiteOpenHelper(Context context, String name, SQLiteDatabase.CursorFactory factory, int version,
                               DatabaseErrorHandler errorHandler) {
        super(context, name, factory, version, errorHandler);
        initWal();
    }

    private void initWal() {
        setWriteAheadLoggingEnabled(true);
    }
}

File: gto-support-db/src/main/java/org/ccci/gto/android/common/db/WalSQLiteOpenHelper.kt
package org.ccci.gto.android.common.db

import android.content.Context
import android.database.DatabaseErrorHandler
import android.database.sqlite.SQLiteDatabase.CursorFactory
import android.database.sqlite.SQLiteOpenHelper

abstract class WalSQLiteOpenHelper @JvmOverloads constructor(
    context: Context?,
    name: String?,
    factory: CursorFactory?,
    version: Int,
    errorHandler: DatabaseErrorHandler? = null,
) : SQLiteOpenHelper(context, name, factory, version, errorHandler) {
    init {
        setWriteAheadLoggingEnabled(true)
    }

    final override fun setWriteAheadLoggingEnabled(enabled: Boolean) = super.setWriteAheadLoggingEnabled(enabled)
}
